euth/users/models.py

```diff
@@ -197,6 +197,7 @@ def signup(self, username, email, timezone, commit=True):
         """Update the fields required for sign-up."""
         self.username = username
         self.email = email
-        self.timezone = timezone
+        if timezone in common_timezones:
```

Comment:
The worst case assignment seems to be an empty string. If this is not desired one should not set `blank=True`.